商品詳細表示機能

shivonnu opened the pull request:
# WHAT
商品詳細表示機能の実装
# WHY
最終課題フリマアプリ機能実装の為。

- 商品が売れているかどうかで表示する'soldout画像'と'購入画面に進む'は、購入機能実装が終わった後に実装します。該当するコメントアウトはそのままにしています。
## gyazoリンク
ボタンの表示を修正したので動画上げます（沢山すみません）。
ログインして他人が出品した詳細表示画面動画(再修正後）
https://gyazo.com/5681b42d202cd5e6e3f28e51e1e0fddf
ログインして本人が出品した詳細表示画面動画(再修正後）
https://gyazo.com/feaf9fb147697d9d65026b53eeba8980
ログアウト時詳細表示画面動画
https://gyazo.com/573e7f66b382e8866d7393974fa572c5

>出品者の記述を変更したので念のため動画上げます。
> ログインして他人が出品した詳細表示画面動画(修正後）
> https://gyazo.com/b05b453abd8b51933f6237a1b5023cbf

> ログインして他人が出品した詳細表示画面動画
> https://gyazo.com/5c7cd9bb9f8059edac3ae835048c2044
> ログインして本人が出品した詳細表示画面動画
> https://gyazo.com/cf36227aacf46e835d2741087df83bbb
> Itemテーブルテストコード実施後ターミナル画像(内容は前工程と同じです)
> https://gyazo.com/a54ed239d9aa84d281195f96139056d0

shivonnu committed:
first commit

indexから商品詳細ページに飛ぶようにリンクを貼りました。詳細ページに画像と名前と値段とコメントが表示されるようにshow.html.erbを編集しました。
アソシエーションをする記述をItemモデルに追加、アイテムコントローラーにてアクティブハッシュの内容を変数に入れるように設定しました。

shivonnu committed:
active_hashの引用と表示

Activehashの情報を引用して表示させました。コントローラーに専用の記述をして、Itemsテーブルに各項目のidカラムを追加しました。

shivonnu committed:
マイグレーションファイルの整理

質問してマイグレーションファイルを整理していただきました。

shivonnu committed:
フォームタグとparamsとvalidatesに○○_idを付与

activehashの５項目にフォームタグとparamsとvalidatesに○○_idを付けました。

shivonnu committed:
READMEを○○_idに修正

READMEをactivehashの５項目のみidでデータを扱うために○○_idに修正しました。

shivonnu committed:
ログインと出品してる場合のみ商品の編集と削除が表示されるように編集

showページの一部をif user_signed_in?でログインと出品してる場合のみ表示する、before actionでログインして無いと出品ページにいけないように修正しました。

shivonnu committed:
テストコードのactivehash５項目○○_idに修正

テストコード項目名前を○○_idに修正、全部OK確認しました。終了した項目はコメントアウトしました。

chirikuzu commented on app/controllers/items_controller.rb:
```diff
@@ -1,11 +1,18 @@
+
+  def show
+    @show = Item.find_by(id: params[:id]) 
+    @user = User.find_by(id: @show.user_id)
```
```
@user = User.find_by(id: @show.user_id)
```
こちら商品とユーザーはアソシエーションを組んでいますので、こちらの変数を定義しない形でもユーザーのニックネームを表示させることは可能かと考えられます。
ビューファイル側にも合わせてコメントさせていただいておりますので、ご確認ください。

chirikuzu commented on app/views/items/show.html.erb:
```diff
@@ -16,54 +16,51 @@
         <tr>
           <th class="detail-item">出品者</th>
-          <td class="detail-value"><%= "出品者名" %></td>
+          <td class="detail-value"><%= @user.nickname %></td>
```
```
<td class="detail-value"><%= @user.nickname %></td>
```
こちらユーザーと商品のアソシエーションを利用して呼び出すことが可能です。
Pictweetなどを参考に記述を修正してみましょう。

chirikuzu commented on app/views/items/show.html.erb:
```diff
@@ -78,7 +75,6 @@
-  <%# /商品の概要 %>
 
   <div class="comment-box">
     <form>
```
101行目にもカテゴリー名を表示させる必要のある箇所がございます。
そちら確認して修正しましょう。

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

shivonnu committed:
出品者の変数を@showに変更して統一、101行目の商品カテゴリーを表示に変更

出品者の変数を@showに変更して統一、101行目の商品カテゴリーを表示させるように修正しました。

shivonnu commented:
コメントありがとうございます！
勉強になりました。修正させて頂きましたので再度レビュー依頼させて頂きます！

chirikuzu reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
大変お手数ではございますが、追加でコメントさせていただいておりますのでご確認をお願いいたします。

shivonnu committed:
出品者自身の出品商品購入ボタンが表示されないように編集

出品者自身の出品商品購入ボタンが出ないように、またログインしてなければ編集、削除、購入ボタンが見えないようにif user_signed_in?を使用して修正しました。

shivonnu commented:
コメントありがとうございます！
修正させて頂きましたので、再度コードレビュー依頼させて頂きます！

yupimaru reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！

shivonnu commented:
ありがとうございます！
引き続き頑張ります！
